contracts/pools/weighted/WeightedPool2Tokens.sol

```diff
@@ -243,17 +239,20 @@ contract WeightedPool2Tokens is
         balanceTokenOut = _upscale(balanceTokenOut, scalingFactorTokenOut);
 
         // Update price oracle with the pre-swap balances
+        MiscData memory miscData = _getMiscData();
         bool tokenInIsToken0 = request.tokenIn == _token0;
         _updateOracle(
-            _getMiscData(),
+            miscData,
             request.lastChangeBlock,
             tokenInIsToken0 ? balanceTokenIn : balanceTokenOut,
             tokenInIsToken0 ? balanceTokenOut : balanceTokenIn
         );
 
         if (request.kind == IVault.SwapKind.GIVEN_IN) {
             // Fees are subtracted before scaling, to reduce the complexity of the rounding direction analysis.
-            request.amount = _upscale(_subtractSwapFeeAmount(request.amount), scalingFactorTokenIn);
+            // This is amount - fee amount, so we round up (favoring a higher fee amount).
+            uint256 feeAmount = request.amount.mulUp(miscData.swapFeePercentage);
```

Comment:
What do you think about `_subtractSwapFeeAmount(amount, fee)`, to avoid the inline computation?

Reply:
if we are not re-using it anywhere else, I'd keep it like this for now